rust/operator-binary/src/controller.rs

```diff
@@ -530,7 +543,25 @@ fn build_rolegroup_catalog_config_map(
                 )
                 .build(),
         )
-        .data(cm_hive_data)
+        .data(
+            catalogs
+                .iter()
+                .map(|catalog| {
+                    let catalog_props = catalog
+                        .properties
+                        .iter()
+                        .map(|(k, v)| (k.to_string(), Some(v.to_string())))
+                        .collect::<Vec<_>>();
+                    Ok((
+                        format!("{}.properties", catalog.name),
+                        product_config::writer::to_java_properties_string(
+                            catalog_props.iter().map(|(k, v)| (k, v)),
+                        )
+                        .context(FailedToWriteJavaPropertiesSnafu)?,
+                    ))
+                })
+                .collect::<Result<_>>()?,
+        )
```

Comment:
I think we should put this data into a Secret as it may contain secrets (e.g. s3 credentials, jdbc credentials, keytabs etc).
It would than make it consistent and use a Secret all the time

Reply:
outdated